### PR TITLE
Batch remove job

### PR DESCRIPTION
**Pull request by Mykhailenko**

### CODE REVIEW

File: scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/common/util/SchedulerProxyUserInterface.java [hunk 1]
```diff
@@ -422,7 +422,12 @@ public boolean pauseJob(JobId jobId) throws NotConnectedException, UnknownJobExc
     public boolean removeJob(JobId jobId) throws NotConnectedException, UnknownJobException, PermissionException {
         checkSchedulerConnection();
         return uischeduler.removeJob(jobId);
+    }
 
+    @Override
+    public boolean removeJobs(List<JobId> jobIds) throws NotConnectedException, PermissionException {
```
Comment (mcompastie) [inline scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/common/util/SchedulerProxyUserInterface.java]: No more  UnknownJobException exception management ?

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java [hunk 2]
```diff
@@ -418,6 +418,19 @@ public boolean removeJob(String sessionId, String jobId) throws RestException {
         }
     }
 
+    @Override
+    public boolean removeJobs(String sessionId, List<String> jobsId) throws RestException {
+        try {
+            // checking permissions
+            checkAccess(sessionId, "DELETE jobs/");
+
+            Scheduler s = checkAccess(sessionId, "DELETE jobs/" + jobsId.get(0));
```
Comment (mcompastie) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java]: You should check if jobsId is empty or not.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java [hunk 3]
```diff
@@ -61,64 +69,64 @@ private boolean isInFinishedState(InternalJob job) {
...
                 start = System.currentTimeMillis();
-                logger.info("Removing job " + jobId);
+                logger.info("Removing jobs " + jobIds.stream().map(JobId::value).collect(Collectors.joining(", ")));
             }
 
             SchedulerDBManager dbManager = service.getInfrastructure().getDBManager();
 
-            List<InternalJob> jobs = dbManager.loadJobWithTasksIfNotRemoved(jobId);
-            TerminationData terminationData;
+            List<InternalJob> jobs = dbManager.loadJobWithTasksIfNotRemoved(jobIds.toArray(new JobId[0]));
```
Comment (mcompastie) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java]: Call me paranoid, but JobId length should be check here as well.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java [hunk 4]
```diff
@@ -61,64 +69,64 @@ private boolean isInFinishedState(InternalJob job) {
...
-            service.getListener()
-                   .jobStateUpdated(job.getOwner(),
-                                    new NotificationData<JobInfo>(SchedulerEvent.JOB_REMOVE_FINISHED,
-                                                                  new JobInfoImpl((JobInfoImpl) job.getJobInfo())));
-
             service.wakeUpSchedulingThread();
         } catch (Exception e) {
-            logger.error("Error while removing job " + jobId, e);
+            logger.error("Error while removing list of jobs " +
+                         jobIds.stream().map(JobId::value).collect(Collectors.joining(", ")), e);
```
Comment (mcompastie) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java]: Putting e.getMessage() would ease the debugging.

File: scheduler/scheduler-server/src/test/java/functionaltests/db/schedulerdb/TestJobRemove.java [hunk 5]
```diff
@@ -179,6 +182,29 @@ public void testFullDataRemove1() throws Exception {
         checkAllEntitiesDeleted();
     }
 
+    @Test
+    public void testMultipleDataRemoved() throws Exception {
+        TaskFlowJob jobDef = new TaskFlowJob();
+        JavaTask task1 = new JavaTask();
+        task1.setName("task1");
+        task1.setExecutableClassName(TestDummyExecutable.class.getName());
+        jobDef.addTask(task1);
```
Comment (mcompastie) [inline scheduler/scheduler-server/src/test/java/functionaltests/db/schedulerdb/TestJobRemove.java]: Regarding the few lines bellow, is the task launched 5 times ?

File: scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/common/util/SchedulerProxyUserInterface.java (hunk 1 above)
Comment (Mykhailenko) [reply scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/common/util/SchedulerProxyUserInterface.java]: Exactly. It was by design. Fabien asked for it.

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java (hunk 2 above)
Comment (Mykhailenko) [reply rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java]: oh. you are right. thanks

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java (hunk 3 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java]: YOU ARE PARANOID!!
(yeah i will check it as well)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java (hunk 4 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java]: Thanks

File: scheduler/scheduler-server/src/test/java/functionaltests/db/schedulerdb/TestJobRemove.java (hunk 5 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/functionaltests/db/schedulerdb/TestJobRemove.java]: Yes, it is submitting the same job 5 times

File: scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/common/util/SchedulerProxyUserInterface.java (hunk 1 above)
Comment (fviale) [reply scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/common/util/SchedulerProxyUserInterface.java]: Additionnally, regarding PermissionException. What if one job cannot be removed due to permission? Is the request still applied to other jobs (notably jobids which are after the offending id)?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/ScriptData.java [hunk 6]
```diff
@@ -46,7 +46,7 @@
 
 
 @Entity
-@NamedQueries({ @NamedQuery(name = "deleteScriptData", query = "delete from ScriptData where taskData.id.jobId = :jobId"),
+@NamedQueries({ @NamedQuery(name = "deleteScriptData", query = "delete from ScriptData where taskData.id.jobId in :ids"),
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/ScriptData.java]: don't we have already the deleteScriptDataInBulk for this purpose ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SelectionScriptData.java [hunk 7]
```diff
@@ -43,7 +43,7 @@
 
 
 @Entity
-@NamedQueries({ @NamedQuery(name = "deleteSelectionScriptData", query = "delete from SelectionScriptData where taskData.id.jobId = :jobId"),
+@NamedQueries({ @NamedQuery(name = "deleteSelectionScriptData", query = "delete from SelectionScriptData where taskData.id.jobId in :ids"),
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SelectionScriptData.java]: same comment as above

**Comment by fviale** [comment]
Looks good in general. would it be possible to double-check manually that the house keeping mechanism is not broken by these changes (unfortunately, we do not have a regression test for this feature).

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/ScriptData.java (hunk 6 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/ScriptData.java]: you are right

File: scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/common/util/SchedulerProxyUserInterface.java (hunk 1 above)
Comment (Mykhailenko) [reply scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/common/util/SchedulerProxyUserInterface.java]: Currently, one permission exception would abort whole operation

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SelectionScriptData.java (hunk 7 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SelectionScriptData.java]: yeah, reused existing

**Pull request merged**